### PR TITLE
frmw821-replacesbstabs

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: https://jira.rax.io/browse/FRMW-821

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM
- [x] QE LGTM

### CODE REVIEW

File: src/elements/Tabs/docs/tabs.docs.html [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <p>
-  This component provides styles and a demo for the
-  <a href="http://angular-ui.github.io/bootstrap/#/tabs">the Angular-UI Bootstrap Tabs plugin</a>,
```
Comment (cathysiller) [inline src/elements/Tabs/docs/tabs.docs.html]: @CITguy any feedback on how/what we should refactor this text to considering the removal of Bootstrap? Thanks!

File: src/elements/Tabs/docs/Tabs.meta.json [hunk 2]
```diff
@@ -2,8 +2,10 @@
     "displayName": "Tabs",
     "stability": "stable",
     "description": "Styling for ngBootstrap Tabs plugin",
```
Comment (cathysiller) [inline src/elements/Tabs/docs/Tabs.meta.json]: does this line need to be updated @CITguy?

**Comment by jj4th** [comment]
QE LGTM from a testing standpoint, doesn't look like much changed there.

File: src/elements/Tabs/docs/Tabs.meta.json (hunk 2 above)
Comment (CITguy) [inline src/elements/Tabs/docs/Tabs.meta.json]: Yeah, we'd better change it now or we'll forget to do it later.

File: src/elements/Tabs/docs/tabs.docs.html (hunk 1 above)
Comment (CITguy) [inline src/elements/Tabs/docs/tabs.docs.html]: Maybe...
```
This element provides styles and behavior for horizontal tabs brought in from the Angular-UI Bootstrap project.
```

File: src/elements/Tabs/scripts/rxTabs.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
+angular.module('encore.ui.elements')
+/**
+ * @ngdoc controller
+ * @name elements.controller:rxTabsetController
+ * @description
+ * Controller for creating tabs.
+ */
+.controller('rxTabsetController', ['$scope', function ($scope) {
```
Comment (CITguy) [inline src/elements/Tabs/scripts/rxTabs.js]: Please separate each angular asset into their own file.

* **rxTabsetController**: `src/utilities/rxTabsetController/scripts/rxTabsetController.js`
  * will require setting up the rxTabsetController utility page
* **rxTabset**: `src/elements/Tabs/scripts/rxTabset.js`
* **rxTab**: `src/elements/Tabs/scripts/rxTab.js`
* **rxTabHeadingTransclude**: `src/elements/Tabs/scripts/rxTabHeadingTransclude.js`
* **rxTabContentTransclude**: `src/elements/Tabs/scripts/rxTabContentTransclude.js`

File: utils/rx-page-objects/test/rxTabs.midway.ts [hunk 4]
```diff
@@ -3,17 +3,17 @@
 import {expect} from 'chai';
 import {$} from 'protractor';
 
-import {Tab, Tabs} from '../index';
+import {rxTabs, Tab} from '../index';
```
Comment (CITguy) [inline utils/rx-page-objects/test/rxTabs.midway.ts]: Do we still need `Tab`?

File: src/elements/Tabs/docs/examples/tabs.simple.html [hunk 5]
```diff
@@ -1,42 +1,42 @@
...
-      you ain't nothin', don't believe him And if he can't learn to love you,
-      you should leave him Cause sister, you don't need him And I ain't tryin'
-      to gas ya up, I just call 'em how I see 'em
-    </p>
-  </tab>
-</tabset>
+  </rx-tab>
+  <rx-tab>
+    <rx-tab-heading>Cat Ipsum <span class="subdued">(meow)</span></rx-tab-heading>
+      <p>
```
Comment (CITguy) [inline src/elements/Tabs/docs/examples/tabs.simple.html]: Extra indentation for the paragraph is unnecessary. Please revert.

File: src/elements/Tabs/docs/examples/tabs.simple.html [hunk 6]
```diff
@@ -1,42 +1,42 @@
...
+        goofballs hide when guests come over. Hide when guests come over intrigued
+        by the shower, or stare at ceiling climb leg for stretch and use lap as
+        chair. Hunt anything that moves flop over, leave dead animals as gifts for
+        chase imaginary bugs, chase mice for stare at ceiling yet hopped up on
+        goofballs.
+      </p>
+  </rx-tab>
+  <rx-tab>
+    <rx-tab-heading>Tupac Ipsum</rx-tab-heading>
+      <p>
```
Comment (CITguy) [inline src/elements/Tabs/docs/examples/tabs.simple.html]: Extra indentation for the paragraph is unnecessary. Please revert.

Comment (cathysiller) [inline src/elements/Tabs/docs/examples/tabs.simple.html]: reverted

File: src/elements/Tabs/docs/examples/tabs.simple.html (hunk 5 above)
Comment (cathysiller) [inline src/elements/Tabs/docs/examples/tabs.simple.html]: reverted

File: src/elements/Tabs/scripts/rxTabs.js (hunk 3 above)
Comment (cathysiller) [inline src/elements/Tabs/scripts/rxTabs.js]: @CITguy, by utility page do you mean js docs? for rxTabsetController

File: src/elements/Tabs/docs/Tabs.meta.json (hunk 2 above)
Comment (cathysiller) [inline src/elements/Tabs/docs/Tabs.meta.json]: updated

File: src/elements/Tabs/docs/tabs.docs.html (hunk 1 above)
Comment (cathysiller) [inline src/elements/Tabs/docs/tabs.docs.html]: updated

File: utils/rx-page-objects/test/rxTabs.midway.ts (hunk 4 above)
Comment (cathysiller) [inline utils/rx-page-objects/test/rxTabs.midway.ts]: @CITguy, I'm getting failed tests when I remove `Tab` and am looking into why.

File: src/elements/Tabs/scripts/rxTabsetController.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        });
+        selectedTab.active = true;
+        // only call select if it has not already been called
+        if (!selectedTab.selectCalled) {
+            selectedTab.onSelect();
+            selectedTab.selectCalled = true;
+        }
+    };
+
+    ctrl.addTab = function addTab (tab) {
```
Comment (CITguy) [inline src/elements/Tabs/scripts/rxTabsetController.js]: Is is necessary to name the function `addTab`?

File: src/elements/Tabs/scripts/rxTabsetController.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        if (tabs.length === 1 && tab.active !== false) {
+            tab.active = true;
+        } else if (tab.active) {
+            ctrl.select(tab);
+        } else {
+            tab.active = false;
+        }
+    };
+
+    ctrl.removeTab = function removeTab (tab) {
```
Comment (CITguy) [inline src/elements/Tabs/scripts/rxTabsetController.js]: Is is necessary to name the function `removeTab`?

File: src/elements/Tabs/docs/examples/tabs.simple.html [hunk 9]
```diff
@@ -1,34 +1,34 @@
...
+  </p>
+  </rx-tab>
+  <rx-tab heading="Veggie Ipsum">
+  <p>
+    Veggies sunt bona vobis, proinde vos postulo esse magis grape pea sprouts
+    horseradish courgette maize spinach prairie turnip jícama coriander
+    quandong gourd broccoli seakale gumbo. Parsley corn lentil zucchini
+    radicchio maize burdock avocado sea lettuce. Garbanzo tigernut earthnut
+    pea fennel.
+  </p>
```
Comment (cathysiller) [inline src/elements/Tabs/docs/examples/tabs.simple.html]: updated

File: src/elements/Tabs/scripts/rxTabHeadingTransclude.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * @restrict A
+ * @description
+ * Element for transcluding tab heading.
+ */
+.directive('rxTabHeadingTransclude', function () {
+    return {
+        restrict: 'A',
+        require: '^rxTab',
+        link: function (scope, elm) {
+            scope.$watch('headingElement', function updateHeadingElement (heading) {
```
Comment (cathysiller) [inline src/elements/Tabs/scripts/rxTabHeadingTransclude.js]: removed

File: src/elements/Tabs/scripts/rxTabsetController.js (hunk 7 above)
Comment (cathysiller) [inline src/elements/Tabs/scripts/rxTabsetController.js]: removed

File: src/elements/Tabs/scripts/rxTabsetController.js (hunk 8 above)
Comment (cathysiller) [inline src/elements/Tabs/scripts/rxTabsetController.js]: removed

File: src/utilities/rxTabsetController/scripts/rxTabsetController.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
+angular.module('encore.ui.utilities')
+/**
+ * @ngdoc controller
+ * @name utilities.controller:rxTabsetController
+ * @description
+ * Controller for creating tabs.
+ */
+.controller('rxTabsetController', ['$scope', function ($scope) {
```
Comment (parlarjb) [inline src/utilities/rxTabsetController/scripts/rxTabsetController.js]: Remove `['$scope',`

File: src/utilities/rxTabsetController/scripts/rxTabsetController.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+            var newActiveIndex = index === tabs.length - 1 ? index - 1 : index + 1;
+            ctrl.select(tabs[newActiveIndex]);
+        }
+        tabs.splice(index, 1);
+    };
+
+    $scope.$on('$destroy', function () {
+        destroyed = true;
+    });
+}]);
```
Comment (parlarjb) [inline src/utilities/rxTabsetController/scripts/rxTabsetController.js]: `]`

Comment (cathysiller) [inline src/utilities/rxTabsetController/scripts/rxTabsetController.js]: removed

File: src/utilities/rxTabsetController/scripts/rxTabsetController.js (hunk 11 above)
Comment (cathysiller) [inline src/utilities/rxTabsetController/scripts/rxTabsetController.js]: removed

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**